app/lib/observers/midnight_observer.dart

```diff
@@ -12,10 +12,19 @@ void initializeMidnightObserver(WidgetRef ref) {
   final authStream = ref.read(authRepositoryProvider).authStateChangesStream;
 
   StreamSubscription? sub;
-  authStream.listen((newUser) {
+  authStream.listen((newUser) async { // Add async to allow use of await
+
+
+  while (newUser == null) {
+    await Future.delayed(Duration.zero);
+  }
+
+  if (newUser != null) {
```

Comment:
You can put sub?.cancel into the same if branch as the other if (newUser != null) check and don't need to restart.